Mathlib/CategoryTheory/Limits/Shapes/SingleObj.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Limits.Types
+import Mathlib.CategoryTheory.SingleObj
+import Mathlib.GroupTheory.GroupAction.Basic
+
+/-!
+# (Co)limits of functors out of `SingleObj M`
+
+We characterise (co)limits of shape `SingleObj M`. Currently only in the category of types.
+
+## Main results
+
+* `SingleObj.Types.limitEquivFixedPoints`: The limit of `J : SingleObj G ⥤ Type u` is the fixed
+  points of `J.obj (SingleObj.star G)` under the induced action.
+
+* `SingleObj.Types.colimitEquivQuotient`: The colimit of `J : SingleObj G ⥤ Type u` is the
+  quotient of `J.obj (SingleObj.star G)` by the induced action.
+
+-/
+
+universe u v
+
+namespace CategoryTheory
+
+namespace Limits
+
+namespace SingleObj
+
+variable {M G : Type v} [Monoid M] [Group G]
+
+/-- The induced `G`-action on the target of `J : SingleObj G ⥤ Type u`. -/
+instance (J : SingleObj M ⥤ Type u) : MulAction M (J.obj (SingleObj.star M)) where
+  smul g x := J.map g x
+  one_smul x := by
+    show J.map (𝟙 _) x = x
+    simp only [FunctorToTypes.map_id_apply]
+  mul_smul g h x := by
+    show J.map (g * h) x = (J.map h ≫ J.map g) x
+    rw [← SingleObj.comp_as_mul]
+    simp only [FunctorToTypes.map_comp_apply, types_comp_apply]
+    rfl
+
+section Limits
+
+variable (J : SingleObj M ⥤ Type u)
+
+/-- The equivalence between sections of `J : SingleObj M ⥤ Type u` and fixed points of the
+induced action on `J.obj (SingleObj.star M)`. -/
+def Types.sections.equivFixedPoints :
```

Comment:
Does `@[simps]` generate useful simp-lemmas for this def? Idem for other defs...

Reply:
For the ones about limits, yes. For the ones about colimits: Not sure, but the alternative simp lemma that I came up with is: `∀ x : J.obj (SingleObj.star G), Types.colimitEquivQuotient J (colimit.ι J (SingleObj.star G) x) = ⟦x⟧` is solved by simp if I add `@[simps!]` to `colimitEquivQuotient`, so I suppose simply adding `@[simps!]` is fine?

Reply:
If you don't like the simp-lemmas that are auto-generated, you can of course manually prove better ones. But if you are reasonably happy with what `simps!` does, then I suggest we just move on. We can always come back and state manual simp-lemmas.